src/main/java/org/launchcode/mentalhealthcareaccess/models/Provider.java

```diff
@@ -0,0 +1,106 @@
+package org.launchcode.mentalhealthcareaccess.models;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import javax.persistence.Entity;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+
+@Entity
+public class Provider extends AbstractEntity {
+    private String companyName;
+    private String displayName;
+    private String lastName;
+
+    private String phoneNumber;
+    @NotBlank
+    private String firstName;
+    @NotNull
+    private String email;
+    @NotNull
+    private String pwHash;
+    private static final BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
+    public Provider(){ }
+    public Provider(String displayName, String companyName, String firstName, String lastName, String email, String phoneNumber, String password) {
+        this.companyName = companyName;
+        this.firstName = firstName;
+        this.email = email;
+        this.lastName = lastName;
+        this.phoneNumber = phoneNumber;
+        this.pwHash = encoder.encode(password);
+        this.displayName = displayName;
+        }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+    public String getDisplayName() {
+        if (firstName != null && lastName != null) {
+            return displayName = lastName + ", " + firstName;
+        }
+        if (companyName != null){
+            return displayName = companyName;
+        }
+        return displayName;
```

Comment:
If this line of code is reached, `displayName` won't be defined (and the compiler may actually throw and error). One way to adjust this is to first declare `displayName` before both conditionals, then set `displayName` within the conditionals (rather then returning it).